javascript/javascript3/week2/setTime.js

```diff
@@ -0,0 +1,23 @@
+//Calling this function will return a promise that resolves after the resolveAfter seconds has passed
+function setTime(resolveAfter) {
+    return new Promise((resolve, reject) => {
+      if(typeof resolveAfter !== "number") {
+        reject(`Time passed should be a number`);
+    }
+      setTimeout(resolve, resolveAfter * 1000)
+    });  
+}
+
+//using it with async/await
+async function asyncSetTime () {
+  await setTime(4).then(() => console.log("I am called asynchronously"))
+        .catch((error) => console.log(error));
+  console.log('This logs after the promise is resolved as we used await')
+}
+
+setTime(2).then(() => {
+    console.log("I am called asynchronously"); // logged out after 8 seconds
```

Comment:
I have a small challenge for you:
See if you can figure it out to check if console.log("I am called asynchronously"); logs after 8 seconds, as the comment says.  
Something tells me that it is much less than that. 
Let me know if you have any questions.

Reply:
Hi @ricardoaguiar ,

Yes, it logs after 2 seconds, the comment mentioned after log statement is wrong.